services/src/main/java/org/exoplatform/addons/gamification/service/configuration/ChallengeServiceImpl.java

```diff
@@ -64,7 +82,13 @@ public Challenge createChallenge(Challenge challenge, String username) throws Il
     if (!spaceService.isManager(space, username)) {
       throw new IllegalAccessException("user is not allowed to create challenge");
     }
-    return challengeStorage.saveChallenge(challenge, username);
+    challenge = challengeStorage.saveChallenge(challenge, username);
+    try {
+      listenerService.broadcast(POST_CREATE_CHALLENGE_EVENT, challenge.getId(), null);
```

Comment:
It will be better to stick to the definition of the broadcast function and pass the source , then the modified object and avoid passing a null
```suggestion
      listenerService.broadcast(POST_CREATE_CHALLENGE_EVENT, challenge.getId(), null);
```